src/v/cluster/tests/rm_stm_tests.cc

```diff
@@ -843,49 +843,6 @@ SEASTAR_THREAD_TEST_CASE(async_adl_snapshot_validation) {
     sync_ser_verify(make_tx_snapshot());
     async_ser_verify(make_tx_snapshot());
 
-    auto make_tx_snapshot_v0 = []() {
-        return reflection::tx_snapshot_v0{
-          .fenced = tests::random_frag_vector(model::random_producer_identity),
-          .ongoing = tests::random_frag_vector(model::random_tx_range),
-          .prepared = tests::random_frag_vector(cluster::random_prepare_marker),
-          .aborted = tests::random_frag_vector(model::random_tx_range),
-          .abort_indexes = tests::random_frag_vector(
-            cluster::random_abort_index),
-          .offset = model::random_offset(),
-          .seqs = tests::random_frag_vector(cluster::random_seq_entry_v0)};
-    };
-
-    sync_ser_verify(make_tx_snapshot_v0());
-    async_ser_verify(make_tx_snapshot_v0());
-
-    auto make_tx_snapshot_v1 = []() {
-        return reflection::tx_snapshot_v1{
-          .fenced = tests::random_frag_vector(model::random_producer_identity),
-          .ongoing = tests::random_frag_vector(model::random_tx_range),
-          .prepared = tests::random_frag_vector(cluster::random_prepare_marker),
-          .aborted = tests::random_frag_vector(model::random_tx_range),
-          .offset = model::random_offset(),
-          .seqs = tests::random_frag_vector(cluster::random_seq_entry_v1)};
-    };
-
-    sync_ser_verify(make_tx_snapshot_v1());
-    async_ser_verify(make_tx_snapshot_v1());
-
-    auto make_tx_snapshot_v2 = []() {
-        return reflection::tx_snapshot_v2{
-          .fenced = tests::random_frag_vector(model::random_producer_identity),
-          .ongoing = tests::random_frag_vector(model::random_tx_range),
-          .prepared = tests::random_frag_vector(cluster::random_prepare_marker),
-          .aborted = tests::random_frag_vector(model::random_tx_range),
-          .abort_indexes = tests::random_frag_vector(
-            cluster::random_abort_index),
-          .offset = model::random_offset(),
-          .seqs = tests::random_frag_vector(cluster::random_seq_entry)};
-    };
-
-    sync_ser_verify(make_tx_snapshot_v2());
-    async_ser_verify(make_tx_snapshot_v2());
-
     auto make_tx_snapshot_v3 = []() {
```

Comment:
nit: can we now name it simply `tx_snapshot` ? as we have only one version

Reply:
we still have 2 versions, from recently added dynamic coordinator partitioning changes

```
uint8_t rm_stm::active_snapshot_version() {
    if (is_transaction_partitioning()) {
        return tx_snapshot::version;
    }
    return tx_snapshot_v3::version;
}
```

I have to add another version (unfortunately) because the new state requires a different serialization. But we can clean them all up in a couple of release.